### PR TITLE
Added recipes to extract seeds for some food such as the tomatoes.

### PR DESCRIPTION
**Pull request by mdtrooper**
Hi.

I made a question about this in Cataclysm DDA subreddit and somebody wrote "you could add the recipes to get seeds from the food." https://www.reddit.com/r/cataclysmdda/comments/51s6jb/can_i_get_seeds_from_the_vegetables/

Well, I just did it.

There are not recipes for all seeds (because for example the marihuana can not get seeds but in the game there are two items) and I have chosen a small value for the time and difficult.

Regards.

### CODE REVIEW

**Comment by Xnfinity** [comment]
I would personally suggest changing them to be disassemble/uncraft recipes to not clutter the crafting menu

**Comment by WizardOfOoo** [comment]
I would suggest taking the marloss out, it's sort of a special spoiler thing.  But it's balanced as it is.

**Comment by shedya** [comment]
Can you extract seeds from irradiated food with this?

File: data/json/recipes/food/seed.json [hunk 1]
```diff
@@ -1,5 +1,161 @@
...
+        "result": "marloss_seed",
+        "type": "recipe",
+        "category": "CC_FOOD",
+        "subcategory": "CSC_FOOD_OTHER",
+        "skill_used": "survival",
+        "difficulty": 2,
+        "time": 5000,
+        "autolearn": true,
+        "qualities": [ { "id": "CUT", "level": 1 } ],
+        "components": [ [ [ "marloss_berry", 1 ] ] ]
```
Comment (Coolthulhu) [inline data/json/recipes/food/seed.json]: Marloss is special magic shit and seeds are supposed to be rarer than berries here.

File: data/json/recipes/food/seed.json [hunk 2]
```diff
@@ -1,5 +1,161 @@
...
+        "result": "seed_corn",
+        "type": "recipe",
+        "category": "CC_FOOD",
+        "subcategory": "CSC_FOOD_OTHER",
+        "skill_used": "survival",
+        "difficulty": 2,
+        "time": 5000,
+        "autolearn": true,
+        "qualities": [ { "id": "CUT", "level": 1 } ],
+        "components": [ [ [ "corn", 1 ] ] ]
```
Comment (Coolthulhu) [inline data/json/recipes/food/seed.json]: "corn" is canned corn and thus not suitable for planting.

File: data/json/recipes/food/seed.json [hunk 3]
```diff
@@ -1,5 +1,161 @@
...
+        "result": "seed_canola",
+        "type": "recipe",
+        "category": "CC_FOOD",
+        "subcategory": "CSC_FOOD_OTHER",
+        "skill_used": "survival",
+        "difficulty": 2,
+        "time": 5000,
+        "autolearn": true,
+        "qualities": [ { "id": "CUT", "level": 1 } ],
+        "components": [ [ [ "canola", 1 ] ] ]
```
Comment (Coolthulhu) [inline data/json/recipes/food/seed.json]: Canola items don't spawn anywhere (only seeds do). They should be obsoleted.

File: data/json/recipes/food/seed.json [hunk 4]
```diff
@@ -1,5 +1,161 @@
...
+        "result": "seed_carrot",
+        "type": "recipe",
+        "category": "CC_FOOD",
+        "subcategory": "CSC_FOOD_OTHER",
+        "skill_used": "survival",
+        "difficulty": 2,
+        "time": 5000,
+        "autolearn": true,
+        "qualities": [ { "id": "CUT", "level": 1 } ],
+        "components": [ [ [ "carrot", 1 ] ] ]
```
Comment (Chezzo) [inline data/json/recipes/food/seed.json]: I love this idea! You totally SHOULD be able to get seeds from things like pumpkins and tomatoes.

You don't get carrot seeds from eating carrots. The plants only make seeds if you don't pick them and allow the parts (that you don't eat) to flower and seed.

File: data/json/recipes/food/seed.json [hunk 5]
```diff
@@ -1,5 +1,161 @@
...
+        "type": "recipe",
+        "category": "CC_FOOD",
+        "subcategory": "CSC_FOOD_OTHER",
+        "skill_used": "survival",
+        "difficulty": 2,
+        "time": 5000,
+        "autolearn": true,
+        "qualities": [ { "id": "CUT", "level": 1 } ],
+        "components": [ [ [ "barley", 1 ] ] ]
+    },
```
Comment (Chezzo) [inline data/json/recipes/food/seed.json]: I will ask my Horticulturist wife about this one, but I think they cook the barley before they put it in stores, or before you can eat it, and that renders the barley unplantable.

File: data/json/recipes/food/seed.json [hunk 6]
```diff
@@ -1,5 +1,161 @@
...
+        "type": "recipe",
+        "category": "CC_FOOD",
+        "subcategory": "CSC_FOOD_OTHER",
+        "skill_used": "survival",
+        "difficulty": 2,
+        "time": 5000,
+        "autolearn": true,
+        "qualities": [ { "id": "CUT", "level": 1 } ],
+        "components": [ [ [ "oats", 1 ] ] ]
+    },
```
Comment (Chezzo) [inline data/json/recipes/food/seed.json]: Again, I will ask my wife, but I think oats are cooked (and usually pressed?) so you can't plant them.

Comment (Coolthulhu) [inline data/json/recipes/food/seed.json]: "oats" don't spawn on their own, only as farming drops.

**Comment by mdtrooper** [comment]
Ok, I am open to sugestions. Well the points are:

- [ ] remove marloss seed (for to avoid spoilers)
- [ ] remove corn (because it is canned)
- [ ] remove Canola

are they correct?

**Comment by Mecares** [comment]
For a longterm goal i would prefer the recipes to produce the seeds and "cut vegetable" or "vegetable pieces" that still were usable for cooking or at least the preservation recipes

**Comment by Coolthulhu** [comment]
It would be better to just allow planting vegetables and stop rot for planted vegetables for as long as they're planted. Pretty sure it would be achievable without much extra code.

**Comment by BevapDin** [comment]
> It would be better to just allow planting vegetables and stop rot for planted vegetables for as long as they're planted. Pretty sure it would be achievable without much extra code.

Planted seeds are already excluded from rotting, and seed data is an item slot that can be added to every item. Raw beans are planted directly, there is no bean seeds item.

**Comment by mdtrooper** [comment]
@Mecares When I was working in the json files, I thought as you think.

Maybe I could add some items in the json of food such as with id = pieces_&lt;vegetable&gt; and change some recipes for to use &lt;vegatable&gt; and &lt;piece_vegetable&gt; too. 

What do you think?

**Comment by Coolthulhu** [comment]
> Planted seeds are already excluded from rotting

This means it should be safe to just allow planting carrots, sugar beets etc. without intermediate seed recipe.
So this PR should probably drop those recipes and just add seed data to plantable plants.

File: data/json/recipes/food/seed.json (hunk 2 above)
Comment (Reclusive-reptile) [inline data/json/recipes/food/seed.json]: The item id for canned corn is can_corn, "corn" is an ear of corn or corn kernels

**Comment by Coolthulhu** [comment]
9 days and no news.

In case anyone is wondering: the idea is fine, it's just that some of those would be better removed (for example canola, it's obsolete in non-seed form) or planted like raw beans (directly into the ground, with no crafting step).

**Comment by WizardOfOoo** [comment]
If I could get the user rights, I could finish it.

File: data/json/recipes/food/seed.json (hunk 6 above)
Comment (Rivet-the-Zombie) [inline data/json/recipes/food/seed.json]: Indeed, these 'oats' haven't been processed into oatmeal yet, so they could still be planted.

File: data/json/recipes/food/seed.json (hunk 5 above)
Comment (Rivet-the-Zombie) [inline data/json/recipes/food/seed.json]: This barley item is still raw; we have a separate item for the cooked version.

**Comment by Rivet-the-Zombie** [comment]
>Are the rotten versions of things different items?

Negative; it's just a status that's applied to the base item.

**Comment by WizardOfOoo** [comment]
Is it possible to get seeds from rotted vegetables?
I think that in nature that's how they sprout.

**Comment by Rivet-the-Zombie** [comment]
>Is it possible to get seeds from rotted vegetables?

Yes.

>I think that in nature that's how they sprout.

You are correct.

**Comment by mugling** [comment]
> 9 days and no news.

Still stalled. Could be reopened if updated

**Pull request closed without merging**